source/extensions/filters/network/well_known_names.h

```diff
@@ -41,6 +41,8 @@ class NetworkFilterNameValues {
   const std::string RateLimit = "envoy.filters.network.ratelimit";
   // Redis proxy filter
   const std::string RedisProxy = "envoy.filters.network.redis_proxy";
+  // Smtp proxy filter
+  const std::string SmtpProxy = "envoy.filters.network.smtp_proxy";
```

Comment:
I don't think this is needed; I don't see it used anywhere.

Reply:
Ah it looks like postgres_proxy (which I used as the template for this) uses this to populate dynamic_metadata. That got dropped from smtp_proxy along the way... should smtp_proxy populate dynamic_metadata?

Reply:
postgres_proxy populated metadata to address use case when RBAC filter is programmed to reject for example any write operations. Metadata was used to carry info about type of database transaction. If you do not plan to have such use case, you do not need metadata.